Fix maximize/restore glitch of OpaqueBrowserFrameView in Weston.

msisov opened the pull request:
This patch avoids spamming aura with states unless previous state
has been "minimized" state and makes the wayland client to decide
what bounds to use and send these back to aura, which
calls relayout and updates OpaqueBrowserFrameView.

tonikitoo commented on ui/ozone/platform/wayland/wayland_window.cc:
```diff
@@ -363,25 +366,60 @@ void WaylandWindow::ConvertEventLocationToCurrentWindowLocation(
-                                           bool is_fullscreen) {
+                                           bool is_fullscreen,
+                                           bool is_activated) {
+  // Handle restore state from wayland side: is the state was minimized and
```
typo: is -> if

tonikitoo commented on ui/ozone/platform/wayland/wayland_window.cc:
```diff
@@ -363,25 +366,60 @@ void WaylandWindow::ConvertEventLocationToCurrentWindowLocation(
+  // |is_activated| is set to true.
+  if (is_minimized_ && is_activated) {
+    is_minimized_ = false;
+    was_minimized_ = !is_minimized_;
```
for clarity, I would say was_minimized_ = true here, but your take.

msisov replied:
fixed

msisov committed:
fixup! Handle window minimize/maximize/restore in external window mode.

Refactor Display::OnWindowStateChanged so that WindowTree
gets the proper id and calls WindowTreeClient with it.

msisov committed:
Fix maximize/restore glitch of OpaqueBrowserFrameView in Weston.

This patch avoids spamming aura with states unless previous state
has been "minimized" state and makes the wayland client to decide
what bounds to use and send these back to aura, which
calls relayout and updates OpaqueBrowserFrameView.

tonikitoo commented on ui/ozone/platform/wayland/wayland_window.cc:
```diff
@@ -280,12 +281,12 @@ void WaylandWindow::Minimize() {
   xdg_surface_->SetMinimized();
   connection_->ScheduleFlush();
   is_minimized_ = true;
+  was_minimized_ = !is_minimized_;
```
nit: I rather be explicit here: was_minimized_ = false;

msisov replied:
done

tonikitoo reviewed (approved):
lgtm
